### PR TITLE
Add in NFRs and Glossaries

### PR DESCRIPTION
**Pull request by BobbyZhouZijian**
Feel free to comment or push more points to this PR if you wish.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -298,15 +298,23 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 ### Non-Functional Requirements
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+2.  Should be able to hold up to 1000 tasks without a noticeable sluggishness in performance for typical usage.
+3.  The report is expected to be generated and formatted within 2 seconds.
```
Comment (MarcTzh) [inline docs/DeveloperGuide.md]: as of v1.2 there is no feature that requires the generation of reports

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -298,15 +298,23 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+4.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```
Comment (MarcTzh) [inline docs/DeveloperGuide.md]: interacting with GUI with a mouse* ?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -298,15 +298,23 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+5.  The product is not required to handle the printing of reports.
```
Comment (MarcTzh) [inline docs/DeveloperGuide.md]: Again, no reports yet

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -298,15 +298,23 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+9.  The source code should be open source.
+
 
 *{More to be added}*
 
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
-* **Private contact detail**: A contact detail that is not meant to be shared with others
+* **Task**: A task is a collection of details about a specific task that needs to be done, including description, title and so on
```
Comment (MarcTzh) [inline docs/DeveloperGuide.md]: include date and time

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -298,15 +298,23 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* **Report**: A visualisation of the time analysis for a particular module or project the user is engaged in.
```
Comment (MarcTzh) [inline docs/DeveloperGuide.md]: This is not in v1.2

**Review by MarcTzh — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (BobbyZhouZijian) [reply docs/DeveloperGuide.md]: Should I just delete it or mark it as coming soon?

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (BobbyZhouZijian) [reply docs/DeveloperGuide.md]: got it

**Review by MarcTzh — APPROVED** [review]
LGTM

**Pull request merged**